src/main/java/com/ibm/cloud/sdk/core/security/IamAuthenticator.java

```diff
@@ -31,32 +31,19 @@
  * an access token from the IAM token service via the "POST /identity/token" operation.
  * When the access token expires, a new access token will be fetched.
  */
-public class IamAuthenticator extends TokenRequestBasedAuthenticator<IamToken, IamToken> implements Authenticator {
+public class IamAuthenticator extends IamRequestBasedAuthenticator implements Authenticator {
   private static final String DEFAULT_IAM_URL = "https://iam.cloud.ibm.com";
   private static final String OPERATION_PATH = "/identity/token";
-  private static final String GRANT_TYPE = "grant_type";
-  private static final String REQUEST_GRANT_TYPE = "urn:ibm:params:oauth:grant-type:apikey";
-  private static final String API_KEY = "apikey";
-  private static final String RESPONSE_TYPE = "response_type";
-  private static final String CLOUD_IAM = "cloud_iam";
-  private static final String SCOPE = "scope";
 
   // Properties specific to an IAM authenticator.
-  private String url;
```

Comment:
Most of the rest of the code deletions you see in this class are due to this code being moved to the new base class.